extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -222,6 +227,36 @@ const BLOCK_SRA_COMBINERS_ = [
   },
 ];
 
+/**
+ * Used to manage messages for different fluid ad slots.
+ *
+ * Maps a sentinel value to an object consisting of the impl to which that
+ * sentinel value belongs and the corresponding message handler for that impl.
+ * @type{!Object<string, !{instance: !AmpAdNetworkDoubleclickImpl, connectionEstablished: boolean}>}
+ */
+const fluidListeners = {};
+
+/**
+ * @param {!Event} event
+ * @private
+ */
+function fluidMessageListener_(event) {
+  const data = tryParseJson(getData(event));
+  if (event.origin != SAFEFRAME_ORIGIN || !data || !data['sentinel']) {
+    return;
+  }
+  const listener = fluidListeners[data['sentinel']];
+  if (listener && data['s'] != 'creative_geometry_update') {
+    if (!listener.connectionEstablished) {
+      listener.instance.connectFluidMessagingChannel();
+      listener.connectionEstablished = true;
+    }
+    return;
+  }
+  listener.instance.receiveMessageForFluid_(data);
```

Comment:
Sorry this will throw... you need to make the listener check surround this call as well:

const listener = fluidListeners[data['sentinel']];
if (!listener) {
    dev().warn(TAG, ...);
    return;
}